src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -679,8 +679,13 @@ public static void Beep()
         {
             if (!Console.IsOutputRedirected)
             {
-                ReadOnlySpan<byte> bell = "\u0007"u8; // Windows doesn't use terminfo, so the codepoint is hardcoded.
-                int errorCode = WindowsConsoleStream.WriteFileNative(OutputHandle, bell, useFileAPIs: Console.OutputEncoding.CodePage != UnicodeCodePage);
+                bool isUnicode = Console.OutputEncoding.CodePage == UnicodeCodePage;
+
+                // Windows doesn't use terminfo, so the codepoint is hardcoded.
+                ReadOnlySpan<byte> bell = isUnicode ? stackalloc byte[2] { 7, 0 } : stackalloc byte[1] { 7 };
```

Comment:
```suggestion
                ReadOnlySpan<byte> bell = isUnicode ? [(byte)7, 0] : [(byte)7];
```

Reply:
I've decompiled the code and is such case it's allocating a managed array.

Reply:
Can you share the IL? It should not be allocating, and I don't see it doing so:
[sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAICYCMBYAKAwGYACbUgYVIG9DSHyyMAWUgWQApgB7HlUlADOAVRhQAxjwAmAUwCU9RnQKM1pAEqyI0gPIwUATwDKABwgwAPMEMJZAPlLBZKAQF5Bo8VLmkA/KQA2gDscKQADAC6pCBBwZEA3ErqpCJCstwuKPKkSaqMAL7JpMWmAE5QAG4QduQ4AGzkbGkZWjr6RmYW1rYOpELmMDk0BSUEBUA)

Reply:
Right... where there do you see an array being allocated?

Reply:
> The C# syntax ILSpy is rendering is not actually an array allocation... look at the IL.

I took the C# syntax for granted. On the other hand I see why ILSpy does that.